### ISSUE TITLE
Add `physical_type` attribute to `Quantity`

### ISSUE THREAD

**Issue by namurphy** [astropy/astropy#14265]
### Description

Right now units have a `physical_type` attribute:

```pycon
>>> import astropy.units as u
>>> u.kg.physical_type
PhysicalType('mass')
```

**Comment by mhvk** [comment]
Hmm, not sure this adds much; one and only one obvious way and all that... Also not sure it would help `get_physical_type` much since the `try/except` would still be useful for things like `Column` (in principle, could replace with `hasattr(obj, "unit")`, but then no need for `Quantity.physical_type`.

**Comment by namurphy** [comment]
A couple of my motivations are (with `quantity` as a `Quantity` instance):

 - Getting the physical type from `quantity.unit.physical_type` isn't that obvious of a way to get the physical type, unless you already know that it's there. In particular, `physical_type` doesn't show up in `dir(Quantity)`.
 - Doing `quantity.unit.physical_type` is less readable than `quantity.physical_type`, in particular since the `unit` there doesn't provide any additional information. 
 - Similarly, doing `quantity.physical_type` is simpler than having to do an intermediate call to `get_physical_type`.
 - Conceptually, `Quantity` objects themselves have a physical type, so one might intuitively expect that it would have a `physical_type` attribute.

Anyway, these are my thoughts!  I also find myself doing `quantity.physical_type` a lot and then having to take a moment to figure out why I'm getting an `AttributeError`.

**Comment by eerovaher** [comment]
I agree with what @namurphy wrote: conceptually `Quantity` has a physical type and it can already be accessed as `Quantity.unit.physical_type`. Implementing the simple property proposed here would be convenient for the users. There is precedence for such convenience properties in `astropy` already, e.g.
https://github.com/astropy/astropy/blob/8ecf379a02d108e9215ede3626154e2c3a363450/astropy/coordinates/sky_coordinate.py#L365-L367

**Comment by mhvk** [comment]
I agree that if I think of `x = 10 m`, I associate the physical type with `x` just as much as with its unit. But the nice current thing is that `Quantity` is almost like an `ndarray`; it has just a few additional attributes and methods with obvious meaning: `unit`, `value`, `.to()`, `.to_value()`, and `to_string()`. Now there are more, but most of those I feel should really not be there: `.si`, `.cgs`, and `.nansum`. The more arguable cases are `.equivalencies`, an attribute I added when I created logarithmic quantities -- it makes sense for the implementation though perhaps less so for actual usage, and `.info`, which is needed to work with tables. Anyway, my point is that the threshold is bigger than for `SkyCoord`, since we are adding to a small group of 7 actually meaningful quantity-specific attributes and methods. Is that worth it? Does it help enough? Does this make duck-typing with other quantity implementations even harder? 

Anyway, FWIW, I see `physical_type` as more logical to have than any of `.si`, `.cgs` (and `nansum` should just be deprecated), so in that sense my opinion is not strong. 

But I also think the functional form of `u.get_physical_type(x)` is better than either `x.physical_type` or `unit.physical_type`; it is just not really that tightly connected to require it to be a method on either `Quantity` or `Unit`.

p.s. `SkyCoord` really is a bit of a different beast, as it simply wraps a `frame` and indeed has a `__getattr__` that already allows one to access many of a frame's attributes.  Similarly, most quantity implementations wrap (in our case, subclass) `np.ndarray`, and has all of `ndarray`'s attributes and methods. (Also, I find `SkyCoord` rather bloated, so for me more of an argument *not* to add an extra property... amusingly, in numpy there is definitely quite a bit of regret of having added so many methods to `ndarray`.)

**Comment by namurphy** [comment]
> Does this make duck-typing with other quantity implementations even harder?

Good point!  I had been wondering about potential drawbacks for this beyond the implementation of `Quantity` being longer.

> Anyway, FWIW, I see `physical_type` as more logical to have than any of `.si`, `.cgs` (and `nansum` should just be deprecated), so in that sense my opinion is not strong.

I'll admit to liking `.si` since I can use it without having to think! 🙃 

> But I also think the functional form of `u.get_physical_type(x)` is better than either `x.physical_type` or `unit.physical_type`; it is just not really that tightly connected to require it to be a method on either `Quantity` or `Unit`.

Huh, I personally find `x.physical_type` to be more understandable and intuitive since it shows the connection between `x` and its physical type.  

One thing to do is figure out how much use this feature would get.  If there isn't much demand for it, then it wouldn't be worth implementing.  Are there other people following this issue who would regularly make use of a `physical_type` attribute of `Quantity`?

**Comment by mhvk** [comment]
Indeed, in the end it is about whether this will be useful to people. I have not used physical types much myself (even though I *really* like the new system you made!). A tutorial might get it a bit more exposure. It may also get more usage once typing becomes more widespread -- I like `Quantity['length']` over `Quantity['m']` for being really nice and physical.

**Comment by mhvk** [comment]
p.s. #14267 deprecates `Quantity.nansum`...

**Comment by hamogu** [comment]
> Huh, I personally find x.physical_type to be more understandable and intuitive since it shows the connection between x and its physical type, ...

I agree. I have not used physical types ever, but I was was to use them, I would expect a Quantity to have a physical type!

**Comment by nstarman** [comment]
On a separate note, I also think we should deprecate the .si and .cgs attributes. ``to`` can accept unit systems.

**Comment by mhvk** [comment]
Even though I think `.si` and `.cgs` are not good design (logically it would be something like `.to('si')` indeed), deprecating them would not seem worth it -- they are somewhat convenient and the maintenance burden is not very high.

**Comment by nstarman** [comment]
I'm fine with that. I agree it's harder to remove features than to add unless we have a compelling reason.

**Comment by arthurxvtv** [comment]
Hello! I've read the discussion and couldn't understand for sure if a conclusion has been reached about the viability of the feature. However, i'm interested in this issue and would like to work on it. Is it still open for contribution?
